sdk/push-fcm/src/main/java/com/klaviyo/pushFcm/Extensions.kt

```diff
@@ -0,0 +1,213 @@
+package com.klaviyo.pushFcm
+
+import android.annotation.SuppressLint
+import android.content.Intent
+import android.content.pm.ApplicationInfo
+import android.content.pm.PackageManager
+import android.content.res.Resources
+import android.graphics.drawable.AdaptiveIconDrawable
+import android.net.Uri
+import android.os.Build
+import android.os.Bundle
+import androidx.core.app.NotificationCompat
+import androidx.core.app.NotificationManagerCompat
+import androidx.core.content.PackageManagerCompat
+import androidx.core.content.res.ResourcesCompat
+import com.google.firebase.messaging.CommonNotificationBuilder
+import com.google.firebase.messaging.RemoteMessage
+import com.klaviyo.core.Registry
+
+/**
+ * Parse channel ID or fallback on default
+ */
+val RemoteMessage.channel_id: String
+    get() = this.data[Notification.CHANNEL_ID_KEY] ?: "Default"
+
+/**
+ * Parse channel name or fallback on default
+ */
+val RemoteMessage.channel_name: String
+    get() = this.data[Notification.CHANNEL_NAME_KEY] ?: "Default"
+
+/**
+ * Parse channel description or fallback on default
+ */
+val RemoteMessage.channel_description: String
+    get() = this.data[Notification.CHANNEL_DESCRIPTION_KEY]
+        ?: "Push notifications default channel"
+
+/**
+ * Parse channel importance or fallback on default
+ */
+val RemoteMessage.channel_importance: Int
+    get() = this.data[Notification.CHANNEL_IMPORTANCE_KEY]?.toInt()
+        ?: NotificationManagerCompat.IMPORTANCE_DEFAULT
+
+/**
+ * Parse out notification priority or fallback on default
+ */
+val RemoteMessage.notificationPriority: Int
+    get() = this.data[Notification.NOTIFICATION_PRIORITY]?.toInt()
+        ?: NotificationCompat.PRIORITY_DEFAULT
+
+/**
+ * Determine if the message originated from Klaviyo from the tracking params
+ */
+val RemoteMessage.isKlaviyoMessage: Boolean get() = this.data.containsKey("_k")
+
+/**
+ * Determine if the message is a notification from Klaviyo (as opposed to a silent push)
+ */
+val RemoteMessage.isKlaviyoNotification: Boolean
+    get() = this.isKlaviyoMessage && (title?.let { true } ?: body?.let { true } ?: false)
+
+/**
+ * Parse notification title text
+ */
+val RemoteMessage.title: String? get() = this.data[Notification.TITLE_KEY]
+
+/**
+ * Parse notification body text
+ */
+val RemoteMessage.body: String? get() = this.data[Notification.BODY_KEY]
+
+/**
+ * Parse deep link into a [Uri] if present
+ */
+val RemoteMessage.deepLink: Uri? get() = this.data[Notification.URL_KEY]?.let { Uri.parse(it) }
+
+/**
+ * Parse click action (activity or intent filter)
+ * Click action could be explicitly sent, or we should use ACTION_VIEW if a deep link is sent
+ */
+val RemoteMessage.clickAction: String?
```

Comment:
Question since its kind of up in the air right now: But could there ever be a situation in the future where we have `click_action` and `deep_link` both set? Is the idea here that a deep link on the payload would always overwrite click action? I guess I'm still a bit unclear on the benefits of keeping it around right now

Reply:
This is written to allow both to be set

Reply:
Yup, I totally misread this. Nevermind